src/main/java/com/tstepnik/planner/controller/TaskController.java

```diff
@@ -0,0 +1,62 @@
+package com.tstepnik.planner.controller;
+
+import com.tstepnik.planner.domain.Task;
+import com.tstepnik.planner.service.TaskService;
+import com.tstepnik.planner.service.UserService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/tasks")
+public class TaskController {
+
+    private final TaskService taskService;
+    private final UserService userService;
+
+    public TaskController(TaskService taskService, UserService userService) {
+        this.taskService = taskService;
+        this.userService = userService;
+    }
+
+    @PreAuthorize("hasRole('ADMIN')")
+    @GetMapping
+    public ResponseEntity<List<Task>> getTasks() {
+        return ResponseEntity.ok(taskService.findAll());
+    }
+
+
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping("/mine")
+    public ResponseEntity<List<Task>> getUserTasks() {
+        List<Task> tasks = taskService.getUserTasks();
+        return ResponseEntity.ok(tasks);
+    }
+
+    @PostMapping("/add")
```

Comment:
This mapping is not following REST API rules

Reply:
You mean, remove "/add" ?

Reply:
Correct.